connector/src/main/scala/com/google/cloud/spark/bigquery/SparkBigQueryOptions.scala

```diff
@@ -37,7 +37,7 @@ import scala.collection.JavaConverters._
 import scala.util.Properties
 
 /** Options for defining {@link BigQueryRelation}s */
-  case class SparkBigQueryOptions(
+  case class LegacySparkBigQueryOptions(
```

Comment:
Do we need to keep this for backward compatibility? If field names in java config are the same, can't we remove it?